chef/cookbooks/provisioner/recipes/setup_base_images.rb

```diff
@@ -55,95 +58,128 @@
 end
 
 # PXE config
-["share","lib"].each do |d|
-  next unless ::File.exists?("/usr/#{d}/syslinux/pxelinux.0")
-  bash "Install pxelinux.0" do
-    code "cp /usr/#{d}/syslinux/pxelinux.0 #{tftproot}/discovery/"
-    not_if "cmp /usr/#{d}/syslinux/pxelinux.0 #{tftproot}/discovery/pxelinux.0"
+%w(x86_64).each do |arch|
+  # Make it easy to totally disable/enable ppc64le
+  next unless discovery_arches.include? arch
+
+  directory "#{discovery_dir}/#{arch}/#{pxecfg_subdir}" do
+    recursive true
+    mode 0755
+    owner "root"
+    group "root"
+    action :create
   end
-  break
-end
 
-directory pxecfg_dir do
-  recursive true
-  mode 0755
-  owner "root"
-  group "root"
-  action :create
+  template "#{discovery_dir}/#{arch}/#{pxecfg_subdir}/default" do
+    mode 0644
+    owner "root"
+    group "root"
+    source "default.erb"
+    variables(append_line: "#{append_line} crowbar.state=discovery",
+              install_name: "discovery",
+              initrd: "../initrd0.img",
+              kernel: "../vmlinuz0")
+  end
 end
 
-template pxecfg_default do
-  mode 0644
-  owner "root"
-  group "root"
-  source "default.erb"
-  variables(append_line: "#{append_line} crowbar.state=discovery",
-            install_name: "discovery",
-            initrd: "initrd0.img",
-            kernel: "vmlinuz0")
+if discovery_arches.include? "x86_64"
+  package "syslinux"
+
+  ["share", "lib"].each do |d|
+    next unless ::File.exists?("/usr/#{d}/syslinux/pxelinux.0")
```

Comment:
Lint/DeprecatedClassMethods: `File.exists?` is deprecated in favor of `File.exist?`.